acceptance-tests/https_ext_crt_list.go

```diff
@@ -0,0 +1,274 @@
+package acceptance_tests
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+)
+
+/*
+	Tests support for external certificate lists. Test structure:
+
+	1. Deploy HAProxy with internal certificate A
+	2. Update external certificate list to add certificate B
+	3. Verify that HTTPS requests using certificates A, B are working and C is not working
+	4. Update external certificate list to remove B and add C
+	5. Verify that HTTPS requests using certificates A, C are working and B is not working
+
+*/
+
+var _ = Describe("External Certificate Lists", func() {
+	AfterEach(func() {
+		deleteDeployment()
+	})
+
+	It("Uses the correct certs", func() {
+		opsfileSSLCertificate := `---
+# Ensure HAProxy is in daemon mode (syslog server cannot be stdout)
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/syslog_server?
+  value: "/var/vcap/sys/log/haproxy/syslog"
+# Add CertA as a regular certificate
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/crt_list?/-
+  value:
+    snifilter:
+    - cert_a.haproxy.internal
+    ssl_pem:
+      cert_chain: ((cert_a.certificate))((cert_a.ca))
+      private_key: ((cert_a.private_key))
+
+# Configure external certificate list
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/ext_crt_list?
+  value: true
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/ext_crt_list_file?
+  value: ((ext_crt_list_path))
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/ext_crt_list_policy?
+  value: continue
```

Comment:
Maybe this could be a separate PR? I will be a substantial test: 
* Deploy HAProxy without external certs 
* Verify deployment fails
* Deploy HAProxy an os-conf script to write an external cert somewhere when it boots up
* Verify deployment passes
* Verify that cert is being used correctly

Reply:
ok let's mark that for a follow-up PR. I will add a task for it.